EssentialCSharp.Web/Services/CaptchaService.cs

```diff
@@ -1,40 +1,54 @@
+using System.Text.Json;
+using EssentialCSharp.Web.Models;
 using Microsoft.Extensions.Options;
 
 namespace EssentialCSharp.Web.Services;
 
 public class CaptchaService : ICaptchaService
 {
     private IHttpClientFactory ClientFactory { get; }
+    public CaptchaOptions Options { get; } //Set with Secret Manager.
 
-    public CaptchaService(IHttpClientFactory clientFactory)
+    public CaptchaService(IHttpClientFactory clientFactory, IOptions<CaptchaOptions> optionsAccessor)
     {
         ClientFactory = clientFactory;
+        Options = optionsAccessor.Value;
     }
 
     // Verify captcha. Optionally add overload to pass in remoteIp as in the docs
     // https://docs.hcaptcha.com/#verify-the-user-response-server-side
-    public async Task<HttpResponseMessage> Verify(string secret, string token, string sitekey)
+    public async Task<HCaptchaResult?> Verify(string secret, string response, string sitekey)
     {
         // create post data
         List<KeyValuePair<string, string>> postData = new()
         {
             new KeyValuePair<string, string>("secret", secret),
-            new KeyValuePair<string, string>("response", token),
+            new KeyValuePair<string, string>("response", response),
             new KeyValuePair<string, string>("sitekey", sitekey)
         };
 
-
         return await PostVerification(postData);
     }
 
-    public async Task<HttpResponseMessage> PostVerification(List<KeyValuePair<string, string>> postData)
+    public async Task<HCaptchaResult?> Verify(string response)
+    {
+        string secret = Options.SecretKey ?? throw new InvalidOperationException($"{CaptchaOptions.CaptchaSender} {nameof(Options.SecretKey)} is unexpectedly null");
+        string sitekey = Options.SiteKey ?? throw new InvalidOperationException($"{CaptchaOptions.CaptchaSender} {nameof(Options.SiteKey)} is unexpectedly null");

```

Comment:
Rather than validating the options values here, can we just setup validation when the app starts?

Reply:
Switched to using the required keyword